### PR TITLE
#12 upload to rds

### PR DESCRIPTION
**Pull request by qrafiq1**

### CODE REVIEW

File: pipeline/load.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    try:
+        with connection.cursor() as cur:
+            cur.execute(query, names)
+            result = cur.fetchall()
+            if not result:
+                logging.warning("No botanists found for the provided names.")
+            return {botanist['full_name']: botanist['botanist_id'] for botanist in result}
+    except exceptions.DatabaseError as e:
+        logging.error("Database error while fetching botanist IDs: %s", e)
+        raise
```
Comment (SurinaCS) [inline pipeline/load.py:50]: could we add specific errors to be raised?

File: pipeline/load.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    except Exception as e:
+        logging.error("Unexpected error while fetching botanist IDs: %s", e)
+        raise
+
+
+def get_plant_metric_data() -> pd.DataFrame:
+    """Reads the plant metric data from a CSV file."""
+    try:
+        df = pd.read_csv("./clean_plant_info.csv")
```
Comment (FarihaChoudhury) [inline pipeline/load.py:59]: Nice work, for a future pull request we can swap our reading a csv file to passing in the dictionary.

**Review by SurinaCS — COMMENTED** [review]
nice work, nice use of typehints

File: pipeline/load.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        return conn
+    except KeyError as e:
+        logging.error("%s missing from environment variables.", e)
+        raise
+    except exceptions.OperationalError as e:
+        logging.error("Error connecting to database: %s", e)
+        raise
+    except Exception as e:
+        logging.error("Unexpected error while connecting to database: %s", e)
+        raise
```
Comment (keoghrmj) [inline pipeline/load.py:33]: Love the error catching!

File: pipeline/load.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+            row['recording_taken'],
+            row['last_watered'],
+            botanist_id_mapping.get(row['name']),
+            row['plant_id']
+        ), axis=1).tolist()
+
+    if data_to_insert:
+        try:
+            with connection.cursor() as cur:
+                cur.executemany(query, data_to_insert)
```
Comment (keoghrmj) [inline pipeline/load.py:89]: Nice use of the executemany for efficiency

**Review by keoghrmj — APPROVED** [review]
Lovely code

**Pull request merged**